### DISCUSSION TITLE
Allow inheritance of attributes in derived classes

### DISCUSSION THREAD

**Discussion by silkfire** [dotnet/csharplang#2442] [General] [answered]
Something that has annoyed me awhile in C# is that when you derive a class with attributes (either on the class itself or its virtual properties), the attributes are ignored and not accessible on the child class.

```
[Attribute1]
[Attribute2("Name")]
public class Base {}

public class Child : Base {}   // is not marked as having its parent's attributes
```

An additional nice feature would be if child attributes could be mixed with its parent's or overridden (depending if it's the same attribute or not).

**Comment by HaloFour** [comment]
Is there a particular use case for this?  Any code that depends on interrogating attributes can already scan the type hierarchy for attributes applied to the parent classes, depending on how it might support that.  Having the compiler (or reflection API) make that happen automatically could easily break a lot of existing code.

**Comment by Joe4evr** [comment, marked as answer]
Attributes *are* inherited if the respective type is marked with `[AttributeUsage(Inherited = true)]`. :wink:

**Comment by dsaf** [comment]
The IDE still does not have a proper support for them even after all this time unfortunately.
No wonder attribute usage lead to confusion.

https://github.com/dotnet/roslyn/issues/711
